crates/spirv-std/src/arch/demote_to_helper_invocation_ext.rs

```diff
@@ -0,0 +1,48 @@
+/// Demote fragment shader invocation to a helper invocation. Any stores to
+/// memory after this instruction are suppressed and the fragment does not write
+/// outputs to the framebuffer.
+///
+/// Unlike [super::kill], this does not necessarily terminate the invocation. It
+/// is not considered a flow control instruction (flow control does not become
+/// non-uniform) and does not terminate the block.
+///
+/// - **Required Capabilities** `DemoteToHelperInvocationEXT`
+/// - **Required Extensions** `SPV_EXT_demote_to_helper_invocation`
```

Comment:
Turning these into `#[cfg]` on the method would be nice and is something we should do eventually, but not required right now.

Reply:
Yeah, and I think we'd want to do more than `#[cfg]`, as when it's cfg'ed out, the error will just be "`demote_to_helper_invocation` not found", when ideally the error message would tell you, you need those capabilities and extensions.

Reply:
... as in, doing a custom cfg attribute? we already decided against that in our discussions though, I'm confused

Reply:
I'm not sure how it would be exactly, but I meant more like a macro which just expands to two versions of the function containing the `cfg` and a better message similar to `gpu_only`.